be/src/exprs/vectorized/arithmetic_expr.cpp

```diff
@@ -26,7 +26,8 @@ class VectorizedArithmeticExpr final : public Expr {
         auto l = _children[0]->evaluate(context, ptr);
         auto r = _children[1]->evaluate(context, ptr);
         if constexpr (pt_is_decimal<Type>) {
-            return VectorizedStrictDecimalBinaryFunction<OP, false>::template evaluate<Type>(l, r);
+            // Enable overflow checking in decimal arithmetic
+            return VectorizedStrictDecimalBinaryFunction<OP, true>::template evaluate<Type>(l, r);
```

Comment:
Will this check overflow for all ops： +，-，*，/ ?

Reply:
yes, arithmetic overflow check is required to ensure correctness